dask/optimize.py

```diff
@@ -45,7 +47,13 @@ def fuse(dsk):
     >>> d = {'a': 1, 'b': (inc, 'a'), 'c': (inc, 'b')}
     >>> fuse(d)  # doctest: +SKIP
     {'c': (inc, (inc, 1))}
+    >>> fuse(d, keys=['b'])  # doctest: +SKIP
+    {'b': (inc, 1), 'c': (inc, 'b')}
     """
+    if isinstance(keys, (set, list)):
+        keys = set(keys)
+    elif keys is not None:
+        keys = set([keys])
```

Comment:
A tuple is a valid key, which has become common with the dask containers.

Although passing a single key may seem odd for `fuse`, we should at least try to have a consistent API across dask.  I took another stab at this.  For convenience, `keys` arguments in `dask.optimize` functions may be a single key or a possibly nested list of keys, which mirrors the API for `dask.get` functions.  Additionally, sometimes we work with sets of keys internally, so `dask.optimize` also accepts a set of keys.  Sound reasonable?

I also updated the variable names in this `fuse`, so hopefully it is a little easier to follow.